### DISCUSSION TITLE
Dataloader for mini-batch is only for inductive learning with multiple graph?

### DISCUSSION THREAD

**Discussion by Jeriousman** [pyg-team/pytorch_geometric#4135] [Q&A] [answered]
Not the new nodedataloader, but the DataLoader described in [](https://pytorch-geometric.readthedocs.io/en/latest/notes/batching.html#advanced-mini-batching)ADVANCED MINI-BATCHING in PyG documentation. This is only for explicit multiple graphs for inductive learning? That is how I understood. For transductive learning, as it is a huge one graph, we cant use the dataloader but use NeighborLoader (for sampling) to fit into GPU if I am correct? Can you please verify if I am right? If I am wrong can you please specify some details? 

And for inductive learning, I guess we can use NeighborLoader as well if we want to sample even from a relatively small graph of many graphs? 

Thank you in advance.

**Comment by rusty1s** [comment, marked as answer]
Yes, for learning on large graphs, have a look at `NeighborLoader`, `ClusterLoader`, `GraphSAINTLoader`, etc (all loaders that take in a `data` object instead of a `dataset`). If you want to perform inductive learning, you may first obtain a subgraph only containing training nodes: `train_data = data.subgraph(train_mask)`.

**Reply by Jeriousman** [reply]
Thank you for the verification!
